backend/layers/processing/process_cxg.py

```diff
@@ -70,48 +68,21 @@ def process(
             object_key = f"{key_prefix}/{labeled_h5ad_filename}"
         self.download_from_s3(artifact_bucket, object_key, labeled_h5ad_filename)
 
-        # Get fragment file S3 URL if fragment_artifact_id is provided
+        # Get fragment file S3 URL
         fragment_file_path = None
-        self.logger.info(f"Fragment artifact ID: {fragment_artifact_id}")
-        if fragment_artifact_id:
-            fragment_file_path = self.get_fragment_file_s3_url(fragment_artifact_id, artifact_bucket, current_artifacts)
+        fragment_artifacts = [
+            artifact for artifact in current_artifacts if artifact.type == DatasetArtifactType.ATAC_FRAGMENT
+        ]
+        if fragment_artifacts:
+            fragment_artifact_id = fragment_artifacts[0].id
+            fragment_file_path = f"s3://{artifact_bucket}/{fragment_artifact_id}"
```

Comment:
if you are pulling from the DB then you can use the `DatasetArtifactType.uri`. That creates one less place we are building the uri string.